### PR TITLE
Added support for controlling the generated certificate key size

### PR DESCRIPTION
**Pull request by MathiasMoxis**
- Added new admin setting with the values 2046 and 4096 as options for key bit size. Defaults to 2046.
- Added logic to handle this keysize setting on certificate generation
- Added test

### CODE REVIEW

File: tests/locallib_test.php [hunk 1]
```diff
@@ -123,4 +123,26 @@ public function test_create_certificates_if_locked(): void {
...
+        $host = (new \moodle_url($CFG->wwwroot))->get_host();
+        $path_to_key_file = $CFG->dataroot . "/saml2/" . $host .".pem";
+
+        $generated_priv_key = openssl_pkey_get_details(openssl_pkey_get_private('file://'.$path_to_key_file, $privatekeypass));
+
+        $key_bit_size = $generated_priv_key['bits'];
+
+        $this->assertEquals(4096, $key_bit_size);
+
+    }
```
Comment (sampraxis) [inline tests/locallib_test.php:147]: Can you also add one more test where the key size is less then 2048 bits or set it to null? So that we can verify that it fallback to 2048.

File: setuplib.php [hunk 2]
```diff
@@ -152,34 +159,57 @@ function pretty_print($arr) {
...
+function set_key_args(&$opensslargs, $keysize): void
+{
+    $keysize = (int)$keysize;
+    // Ensure minimum key size of 2048 bits.
+    if ($keysize < 2048) {
+        $keysize = 2048;
+    }
+    $opensslargs['private_key_type'] = OPENSSL_KEYTYPE_RSA;
+    $opensslargs['private_key_bits'] = $keysize;
+}
```
Comment (sampraxis) [inline setuplib.php]: I think that we should avoid using reference type and since this only modified a single argument, this should return the array instead. If you return the values you might also need to change the function name as well.


```suggestion
function set_key_args($opensslargs, $keysize)
{
    $keysize = (int)$keysize;
    // Ensure minimum key size of 2048 bits.
    if ($keysize < 2048) {
        $keysize = 2048;
    }
    $opensslargs['private_key_type'] = OPENSSL_KEYTYPE_RSA;
    $opensslargs['private_key_bits'] = $keysize;
    return $opensslargs;
}
```

The code where you called it.
```php
$opensslargs = set_key_args($opensslargs, $keysize);
```

**Review by sampraxis — CHANGES_REQUESTED** [review]

**Review by sampraxis — APPROVED** [review]
